Bottom: 2c509b0b8faf8bc6aa9dfa9321643471a4793f69
Top:    f874a5bd0eb37ed6a962d7c708ee25b88f9547e7
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 09:57:34 +0800

debug.patch


---

diff --git a/daemon/automount.c b/daemon/automount.c
index a22de2d..f3c2b0a 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -122,7 +122,7 @@ static int do_mkdir(const char *parent, const char *path, mode_t mode)
 		status = statfs(parent, &fs);
 	if ((status != -1 && fs.f_type == (__SWORD_TYPE) AUTOFS_SUPER_MAGIC) ||
 	    contained_in_local_fs(path)) {
-		mode_t mask = umask(0022);
+		mode_t mask = umask((mode_t) 0022);
 		int ret = mkdir(path, mode);
 		void umask(mask);
 		if (ret == -1) {
diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index c1fc528..aa7ae07 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -881,7 +881,8 @@ static int validate_location(unsigned int logopt, char *loc)
 		    !strncmp(ptr, "file:", 5) || !strncmp(ptr, "yp:", 3) ||
 		    !strncmp(ptr, "nis:", 4) || !strncmp(ptr, "nisplus:", 8) ||
 		    !strncmp(ptr, "ldap:", 5) || !strncmp(ptr, "ldaps:", 6) ||
-		    !strncmp(ptr, "sss:", 4) || !strncmp(ptr, "dir:", 4))
+		    !strncmp(ptr, "sss:", 4) || !strncmp(ptr, "dir:", 4) ||
+		    !strncmp(ptr, "hosts:", 4))
 			return 1;
 		error(logopt,
 		      "expected colon delimeter not found in location %s",
